server/src/views.py

```diff
@@ -47,6 +47,8 @@ def agents():
 def agent_detail(agent_id):
     """Agent detail view"""
     agent_info = mongo.db.agents.find_one({"name": agent_id})
+    if not agent_info:
+        return render_template("agent_not_found.html", agent_id=agent_id)
```

Comment:
Does it return a 404 HTTP code?

Reply:
I was considering that also, was kinda on the fence about it because it wouldn't be seen by the user, but it seems more complete to do it that way. I've added that too.